From d933703525c8a12fed0f71000bc43d6b47959fa4 Mon Sep 17 00:00:00 2001
From: Olivier Paroz <github@oparoz.com>
Date: Sun, 25 Jan 2015 16:38:32 +0100
Subject: [PATCH 1/2] Keep the extension in temp files

The file extension helps some applications like ImageMagick to properly
process files
---
 lib/private/tempmanager.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/private/tempmanager.php b/lib/private/tempmanager.php
index a3bb07f..847295f 100644
--- a/lib/private/tempmanager.php
+++ b/lib/private/tempmanager.php
@@ -42,7 +42,7 @@ public function __construct($baseDir, ILogger $logger) {
 	}

 	protected function generatePath($postFix) {
-		return $this->tmpBaseDir . '/oc_tmp_' . md5(time() . rand()) . $postFix;
+		return $this->tmpBaseDir . '/oc_tmp_' . md5(time() . rand()) . '.' . $postFix;
 	}

 	/**

From 48338185e823d6003dc909e9e4151b0c86388ca5 Mon Sep 17 00:00:00 2001
From: Olivier Paroz <github@oparoz.com>
Date: Sun, 25 Jan 2015 19:05:30 +0100
Subject: [PATCH 2/2] Fix the office class The office converter already
 replaces the original extension with '.pdf', so we need to switch the
 extensions of the tmp file instead of appending '.pdf' to it

---
 lib/private/preview/office.php | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/lib/private/preview/office.php b/lib/private/preview/office.php
index 5bd61bd..1f6d977 100644
--- a/lib/private/preview/office.php
+++ b/lib/private/preview/office.php
@@ -31,12 +31,16 @@ public function getThumbnail($path, $maxX, $maxY, $scalingup, $fileview) {
 		shell_exec($exec);

 		//create imagick object from pdf
+		$pdfPreview = null;
 		try{
-			$pdf = new \imagick($absPath . '.pdf' . '[0]');
+			list( $dirname, , , $filename ) = array_values( pathinfo($absPath) );
+			$pdfPreview = $dirname . '/' . $filename . '.pdf';
+
+			$pdf = new \imagick($pdfPreview . '[0]');
 			$pdf->setImageFormat('jpg');
 		} catch (\Exception $e) {
 			unlink($absPath);
-			unlink($absPath . '.pdf');
+			unlink($pdfPreview);
 			\OC_Log::write('core', $e->getmessage(), \OC_Log::ERROR);
 			return false;
 		}
@@ -45,7 +49,7 @@ public function getThumbnail($path, $maxX, $maxY, $scalingup, $fileview) {
 		$image->loadFromData($pdf);

 		unlink($absPath);
-		unlink($absPath . '.pdf');
+		unlink($pdfPreview);

 		return $image->valid() ? $image : false;
 	}
